ts/src/pro/poloniexfutures.ts

```diff
@@ -0,0 +1,741 @@
+        return message;
+    }
+
+    parseWsBalance (response) {
+        //
+        //    {
+        //        currency: 'USDT',
+        //        availableBalance: '4.0000000000',
+        //        timestamp: '1680557568670'
+        //    }
+        //
+        const timestamp = this.safeInteger (response, 'timestamp');
+        const result = {
+            'info': response,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+        };
+        const currencyId = this.safeString (response, 'currency');
+        const code = this.safeCurrencyCode (currencyId);
+        const newAccount = this.account ();
+        newAccount['free'] = this.safeString (response, 'availableBalance');
+        newAccount['used'] = this.safeString (response, 'orderMargin');
+        result[code] = newAccount;
+        return this.safeBalance (result);
+    }
+
+    handleMessage (client, message) {
+        const subject = this.safeString (message, 'subject');
+        // const type = this.safeString (message, 'type');
+        const event = this.safeString (message, 'event');
+        if (event === 'pong') {
+            return this.handlePong (client, message);
+        }
+        const methods = {
+            'received': this.handleOrderBook,
+            'open': this.handleOrderBook,
+            'done': this.handleOrderBook,
+            'ticker': this.handleTicker,
+            // 'trades': this.handleTrade,
+            'match': this.handleTrade,
+            'orderChange': this.handleOrder,
+            'availableBalance.change': this.handleBalance,
+            'orderMargin.change': this.handleBalance,
+        };
+        const method = this.safeValue (methods, subject);
+        if (subject === 'auth') {
+            this.handleAuthenticate (client, message);
+        } else {
+            if (method !== undefined) {
+                return method.call (this, client, message);
+            }
+        }
+    }
+
+    handleAuthenticate (client, message) {
+        //
+        //    {
+        //        success: true,
+        //        ret_msg: '',
+        //        op: 'auth',
+        //        conn_id: 'ce3dpomvha7dha97tvp0-2xh'
+        //    }
+        //
+        const data = this.safeValue (message, 'data');
+        const success = this.safeValue (data, 'success');
+        const messageHash = 'authenticated';
+        if (success) {
+            client.resolve (message, messageHash);
+        } else {
+            const error = new AuthenticationError (this.id + ' ' + this.json (message));
+            client.reject (error, messageHash);
+            if (messageHash in client.subscriptions) {
+                delete client.subscriptions[messageHash];
+            }
+        }
+        return message;
+    }
+
+    ping (client) {
+        return {
+            'type': 'ping',
+            'id': this.milliseconds (),
+        };
+    }
+
+    handlePong (client, message) {
```

Comment:
Here we can probably use the function that already exists in Client.ts

Reply:
What do you mean exactly?